### DISCUSSION TITLE
c-index calculation for survival random forest with competing risk

### DISCUSSION THREAD

**Discussion by orianelam** [kogalur/randomForestSRC#382] [Q&A]
Hi, 

I am using the RandomForestSRC package to develop a Random Survival Forest model with competing risks. The Harrell's _c_-index calculation was clearly define for survival rf but not in the competing risks setting. Can you enlighten me please ?

I do not find the same _c_-index obtained with the predict() function when I do it with get.cindex() function (while it's ok without competing risks).

A exemple :

```
train <- follic[1:500,]
test<-follic[501:541,]

rf <- rfsrc(Surv(time, status) ~ ., train, 
                    ntree=500, mtry=10,  nodesize=15,
                    splitrule = "logrankCR")

pred <- predict.rfsrc(object = rf, test)

1-pred$err.rate[500,] 
#   event.1     event.2 
#   0.5894632   0.5833333

# c-index calculation with the event-specific expected number of years lost 
1-randomForestSRC::get.cindex(test$time,test$status==1,pred$predicted[,1]) 
# 0.586785
```

It is close but not equal...

It is ok without competing risks :

```
follic2 <- follic
follic2$status <- ifelse(follic2$status==1,1,0)

train <- follic2[1:500,]
test<-follic2[501:541,]

rf <- rfsrc(Surv(time, status) ~ ., train, 
                    ntree=500, mtry=10, nodesize=15, 
                    splitrule = "logrank")

pred <- predict.rfsrc(object = rf, test)
1-pred$err.rate[500] 
# 0.5966469
1-randomForestSRC::get.cindex(test$time,test$status,pred$predicted) 
# 0.5966469
```

Thank u very much!

**Comment by ishwaran** [comment]
The get.cindex function is only designed for right-censored survival data where the event of interest is death and will not give correct error rates for competing risk settings where there are more than one event type: that's why the two values do not agree.

You might also find the following vignette helpful for further information about [competing risks](https://www.randomforestsrc.org/articles/competing.html)

**Reply by orianelam** [reply]
Thank you for your reply. I read this vignette before my post and the calculation of the c-index in the competing risks setting isn't presented. 
So, I need to understand what is the err.rate in the rfsrc() function (in the competing risks setting)... Can you enlighten me please ?
Thank you very much.

**Reply by ChuangshiWang** [reply]
Hi. I have a question about get.cindex function. The value obtained from this function is actually the prediction error, correct? The C-index should be 1-get.cindex? When I read the illustration, I was confused about the "c.indxe" being less than 0.5. Thank you.

**Comment by ishwaran** [comment]
If you look at the vignette, you will see that the predicted value that is used in competing risks is the expected number of years lost (cause-j mortality).  This predicted value is used for calculating a type of concordance index (see page 763 from the [paper](https://ishwaran.org/papers/IGKMGL.biostatistics2014.pdf))

**Reply by orianelam** [reply]
Thank you for the paper. 
I'm verry sorry but this sentence of the paper makes me doubt: "To assess prediction performance, we use the concordance index **and the prediction error defined by the integrated Brier score** (BS)".
So rfsrc()$err.rate is 1 - c-index or it is the Brier score (in the context of competing risks) ?
I'm really sorry... !

**Comment by ishwaran** [comment]
As you noticed, the paper discusses two different ways to measure prediction error: the Brier Score and Concordance index.  Only the Concordance index is supported and used in our R package so the error rates you are using are Concordance error rates.

**Reply by orianelam** [reply]
Thank you very much !!
